src/Compilers/Core/Portable/Syntax/SyntaxTree.cs

```diff
@@ -102,6 +102,13 @@ public ParseOptions Options
         /// </summary>
         public abstract Encoding? Encoding { get; }
 
+        /// <summary>
+        /// Useful information about this tree that is stored for source-generator scenarios.  Allows the incremental
+        /// generation framework to compute and cache data once against a tree so it does not have to go back to source
+        /// for untouched trees when other trees in the compilation are modified.
+        /// </summary>
+        private SourceGeneratorSyntaxTreeInfo _sourceGeneratorInfo = SourceGeneratorSyntaxTreeInfo.NotComputedYet;
```

Comment:
> Did you consider giving it aless specific name?

I def did, but fell back into that flip-flopping of: "this is only for source-generators today", vs "maybe other things in the future could benefit...."

What would honestly make a ton of sense would be if we had SyntaxTree cache the DeclTable entry for it, and then allow the decl table entry to have these additional bits, as well as the other bits it holds onto today.  However, this would require making things common between C# and VB and would def be larger in scope.  

So this was basically a way to solve this problem in a very tiny, scoped fashion.  I def think tehre may be more cohesive approaches we coudl take, but they would also be much larger in scope and i figured would also take ages to get into the compiler :)   I'd like to put SG work to bed (this is literally my last PR for this area), so i went with narrow and targetted.

Reply:
>  In fact it seems like we could even expand on this to maintain this field across incremental parses (maybe).

we def could (though it's a bit non-trivial).  The general idea would be to do something similar to how we store the bits for 'ContainsDiagnostics' and 'ContainsAnnotations'.  Nodes could have a bitset provided to them of interesting bits.  As you build the parent nodes, they union those bits together, and you then have all that information available in the root node.  

This then makes incremental fall out trivially.  If we make a new AttributeList/GlobalUsing node we set that bit, and pass it upwards.

I explored this, but it was much larger in scope *and* it's terrifying from a perf perspective to touch the green nodes at all.  We squeezed perf heavily there, and it's a space where you could change a line and easily cause +- large perf percentages.  So i def didn't want to go that far when i'm really ok just storing two bits here.